Source/WPF/MyMoney/Database/CsvStore.cs

```diff
@@ -151,77 +148,64 @@ public virtual DataSet QueryDataSet(string cmd)
-                        var field2Payee = columns[1];
-                        var field3Amount = columns[2];
+                    throw new NotSupportedException("Invalid CSV format expecting 3 columns [Date, Payee, Amount]");
+                }
 
-                        if (total == 0)
-                        {
-                            if (field1Date != "Date" || field2Payee != "Payee" || field3Amount != "Amount")
-                            {
-                                throw new NotSupportedException("Invalid CSV format The fist row is expected to be the header [Date, Payee, Amount]");
-                            }
-                        }
-                        else
-                        {
-                            var dateTokens = field1Date.Split('-');
+                var field1Date = csvReader[0];
+                var field2Payee = csvReader[1];
+                var field3Amount = csvReader[2];
+                if (total == 0)
+                {
+                    if (field1Date != "Date" || field2Payee != "Payee" || field3Amount != "Amount")
+                    {
+                        throw new NotSupportedException("Invalid CSV format The fist row is expected to be the header [Date, Payee, Amount]");
+                    }
+                }
+                else
+                {
+                    var dateTokens = field1Date.Split('-');
 
-                            if (dateTokens.Length != 3)
-                            {
-                                throw new NotSupportedException("Invalid CSV format The Date needs to be specified in ISO8601 YYYY-MM-DD format : " + field1Date);
-                            }
+                    if (dateTokens.Length != 3)
+                    {
+                        throw new NotSupportedException("Invalid CSV format The Date needs to be specified in ISO8601 YYYY-MM-DD format : " + field1Date);
+                    }
 
-                            if (dateTokens[0].Length != 4)
-                            {
-                                throw new NotSupportedException("Invalid CSV format The Date Year must be 4 digits : " + field1Date);
-                            }
+                    if (dateTokens[0].Length != 4)
+                    {
+                        throw new NotSupportedException("Invalid CSV format The Date Year must be 4 digits : " + field1Date);
+                    }
 
-                            Transaction t = myMoney.Transactions.NewTransaction(acct);
+                    Transaction t = myMoney.Transactions.NewTransaction(acct);
 
-                            t.Id = -1;
-                            t.Date = DateTime.Parse(field1Date);
-                            t.Payee = t.Payee = myMoney.Payees.FindPayee(field2Payee, true);
-                            t.Amount = decimal.Parse(field3Amount);
+                    t.Id = -1;
+                    t.Date = DateTime.Parse(field1Date);

```

Comment:
Do you want this to "throw" or would "TryParse" be better?

Reply:
Throw, for now this is a dirty quick importer and I want the user <ME :)> to be educated on the invalid file content.
In the future I need to put more time on this and have a full CSV data analyzer and preview UX